jujugui/static/gui/src/app/jujulib/charmstore.js

```diff
@@ -83,6 +83,10 @@ var module = module;
     _transformQueryResults: function(callback, error, data) {
       if (error !== null) {
         callback(error, data);
+      } else if (!data) {
+        // The only time there should be no data and no error is if the
```

Comment:
I agree.

The `getEntity` call response callback is first wrapped with the `jujulib._wrap` and `charmstore._transformQueryResults` both of which are used by multiple functions. I'm not able to add it to _wrap as it's used by every call so I'd have to create another wrapper. I think that keeping it in _transformQueryResults makes sense because it can't transform data that doesn't exist.